pennylane_lightning/core/src/observables/Observables.hpp

```diff
@@ -414,4 +414,105 @@ class HamiltonianBase : public Observable<StateVectorT> {
     }
 };
 
+/**
+ * @brief Sparse representation of SparseHamiltonian<T>
+ *
+ * @tparam T Floating-point precision.
+ */
+template <class StateVectorT>
+class SparseHamiltonianBase : public Observable<StateVectorT> {
+  public:
+    using PrecisionT = typename StateVectorT::PrecisionT;
+    using ComplexT = typename StateVectorT::ComplexT;
+    // cuSparse required index type
+    using IdxT =
+        typename std::conditional<std::is_same<PrecisionT, float>::value,
+                                  int32_t, int64_t>::type;
```

Comment:
We may want this to be a templated parameter instead, to all it to be implemented according to the needs of the backend. The choice of value here can be defined in the child-class explicitly when inheriting.

Reply:
I would like to keep it as it for now and revisit it later.